### ISSUE TITLE
VMware + vmx issues

### ISSUE THREAD

**Issue by evilezh** [hashicorp/packer#4885]
Platform: Windows 10 pro
VMware: VMware workstation pro 12
Packer: 1.0

1. it works only with ide disks. In case with non-ide drive .. it reports something like: 'e:\output-vmware-vmx\auto detect' not found. I did quick lookup into code - and found out that only IDE is supported (hardcoded). Nothing in documentation.

2. created files names have non-interpolated field - timestamp: packer-vmware-vmx-{{timestamp}}.vmx

Both problems are not critical .. i can rename later files or if you can read code, you can figure out, that only IDE drive supported. I would suggest add scsi/sata support as it is kinda default always rather than IDE.

**Comment by rickard-von-essen** [comment]
Please supply the information requested in the issue template:
- Packer version (`packer version`)
- Host platform (`uname -a` etc.)
- Debug log output from `PACKER_LOG=1 packer build template.json`.
  Please paste this in a [gist](https://gist.github.com).
- The _simplest example template and scripts_ needed to reproduce the bug.
  Include these in your [gist](https://gist.github.com).

---

1) I don't follow this, please be a bit more precise and supply the requested information.

**Comment by evilezh** [comment]
I updated packer version ...
Windows 10 pro is quite specific - there isn't anything else to say. Latest updates, plenty of memory/cpu etc.

I provided enough information.
for 1: here is exact line of code.
https://github.com/hashicorp/packer/blob/9f992b8f809525903049fdda94a4dcdeaeb0cd8a/builder/vmware/common/step_clean_vmx.go#L56
or line 46 where IDE only regex defined. Maybe something like 
```golang
devRe := regexp.MustCompile(`^(ide|scsi|sata)\d:\d\.`)
```
would be enough ? Not sure about scsi/sata patterns.

for 2: there is nothing to output, that is exact name of resulting file: packer-vmware-vmx-{{timestamp}}.vmx

here is last lines
```
==> vmware-vmx: Gracefully halting virtual machine...
    vmware-vmx: Waiting for VMware to clean up after itself...
==> vmware-vmx: Deleting unnecessary VMware files...
    vmware-vmx: Deleting: output-vmware-vmx\packer-vmware-vmx-{{timestamp}}.vmx.lck\M29079.lck
    vmware-vmx: Deleting: output-vmware-vmx\vmware.log
==> vmware-vmx: Compacting the disk image
==> vmware-vmx: Cleaning VMX prior to finishing up...
    vmware-vmx: Unmounting floppy from VMX...
    vmware-vmx: Disabling VNC server...
Build 'vmware-vmx' finished.
```
here is config section:
```
{
      "type":             "vmware-vmx",
      "source_path":      "C:\\Users\\*******\\Documents\\Virtual Machines\\ubuntu-server\\ubuntu-server.vmx",
      "ssh_username":     "ubuntu",
      "ssh_password":     "*******",
      "shutdown_command": "sudo shutdown -P 0"
    }
```
I used SCSI drives in vmware which are by default.

**Comment by znerd** [comment]
I've got the same issue where `{{timestamp}}` is literally in the file name, with Packer 1.0.0 and VMware Fusion Pro 8.5.7 (provider `vmware-vmx`).

Note that I'm using VirtualBox (provider `virtualbox-ovf`) as well, and that one works flawless.

**Comment by charlievieth** [comment]
I ran into this error as well.  It comes from `vmware-vdiskmanager`, which packer shells out to, during the `CompactDisk` step.  I was able to fix it by changing `ide0:0.fileName = "auto detect"` to `ide0:0.autodetect = "TRUE"` in the target vmx file.  It would appear that VMware Workstation and Fusion use a mix of the two to indicate that a CD is set to Auto-detect.

**Comment by tiefox** [comment]
Same issue for me with packer 1.0.2 and Vmware fusion 7.1.3.

All files have {{timestamp}} in the name.
But I dont have any ide0:0, only sata and is set already to "TRUE".

**Comment by mwhooker** [comment]
the timestamp problem should be fixed in #5274

**Comment by goodwinb99** [comment]
I'm having the same problem with it trying to compact a non-existent file because I can't find a way to *delete* a key in vmx_data using packer.  I need to delete, e.g., ide0:0.filename because if that value is set to anything, it'll try to compact it, regardless of the autodetect setting.  Is there a way to, in packer, remove a key from the vmx file when the builder is done?

**Comment by Selora** [comment]
Same issue as @rastating  & others. Same fix.
`ide0:0.fileName = "auto detect"` -> `ide0:0.autodetect = "TRUE"`
-Packer 1.2.0
-VMware Fusion 10 Pro

**Comment by eduncan911** [comment]
Confirmed same issue.  Building an ubuntu image on Fusion using `vmware-iso` to generate a base vm, and then using `vmware-vmx` to create additional VMs from that base vmx.

I fixed it my changing my packer template to have the `ide0:0.autodetect: true` as seen below:
```
{
  "builders": [
    {
      "type": "vmware-iso",
      ...snip...
      "vmx_data": {
        ...snip...
        "ide0:0.autodetect": true
      },
     ...
```

This then removed the ClientDevice setting and instead made this in my source VMX:

```
ide0:0.autodetect = "1"
ide0:0.devicetype = "cdrom-image"
ide0:0.filename = "/Users/ericduncan/code/packer-test/local_files/ubuntu-14.04.5-server-amd64.iso"
ide0:0.present = "TRUE"
```

**Comment by Mrmayank** [comment]
@eduncan911 The above fix didn't work in builder 'vmware-iso' for Ubuntu. That image is built but when vmware-vmx is built using that image the same error came  'some_dir\auto detect' : A file was not found (0x4).'

**Comment by DanHam** [comment]
Hi all. Hopefully the reported errors at the compacting stage of the VMX build should be fixed by #6074. Please see the comment [here](https://github.com/hashicorp/packer/pull/6074#issuecomment-378601186) for further info.

**Comment by SwampDragons** [comment]
If anyone would be willing to try the patch in #6074 and verify that it solves some of the problems here, let me know your system architecture and I can build you a binary.

**Comment by embusalacchi** [comment]
I will try the patch.

**Comment by DanHam** [comment]
@embusalacchi Thanks for offering to test!

Do you need a binary built or are you able to build the binary + patch yourself? If not what OS do you need the Packer binary for? @SwampDragons should be able to provide one for you.

**Comment by embusalacchi** [comment]
@DanHam OSX That would be super helpful - OSX binary would be great.

**Comment by DanHam** [comment]
@SwampDragons Could you provide @embusalacchi with the required binary?

**Comment by SwampDragons** [comment]
@embusalacchi here you go: 
[packer.zip](https://github.com/hashicorp/packer/files/1904149/packer.zip)

**Comment by embusalacchi** [comment]
@SwampDragons @DanHam I tested the binary with the same workflow that failed for me yesterday with the error and it worked without a problem!

**Comment by DanHam** [comment]
@embusalacchi Great! Thanks for testing. Were you able to test with a VM with multiple disks?

**Comment by embusalacchi** [comment]
@DanHam I did not test with multiple disks but it is easy for me to do - I can do that tomorrow morning (EST) and let you know.

**Comment by embusalacchi** [comment]
A little late but I did a run with multiple disks and it worked!

**Comment by DanHam** [comment]
@embusalacchi Better late than never! Thanks a lot for helping out.

**Comment by SwampDragons** [comment]
Closed in #6074
